### PR TITLE
Signatures plugin: Fix regex for matching bbcode image tags.

### PR DESCRIPTION
**Pull request by beckyvb**
Fixes bug where signatures could have more signatures than allowed if no spaces were between the multiple image tags.

### CODE REVIEW

**Comment by linc** [comment]
Why is the version jumping from 1.5.6 to 1.6.1?

**Comment by beckyvb** [comment]
It's currently on version 1.6.0 on stage. I wasn't sure how to deal with the discrepancy. Agree that hopping from 1.5.6 to 1.6.1 does seem somewhat disingenuous.

**Comment by linc** [comment]
Master is the canonical version, so I think 1.5.7 would be correct.

**Comment by tburry** [comment]
This really should not be checking against BBCode. It should be formatting whatever signature into HTML and then checking the number of images there. Otherwise it will fail for Markdown or whatever other format we end up supporting.

File: plugins/Signatures/class.signatures.plugin.php [hunk 1]
```diff
@@ -267,7 +267,8 @@ public function CheckSignatureLength($Values, &$Sender) {
    public function CheckNumberOfImages($Values, &$Sender) {
       if (C('Plugins.Signatures.MaxNumberImages') && C('Plugins.Signatures.MaxNumberImages') !== 'Unlimited') {
          $max = C('Plugins.Signatures.MaxNumberImages');
-         $numMatches = preg_match_all('/(<img|\[img.*\]|\!\[.*\])/i', $Values['Plugin.Signatures.Sig']);
+         $Sig = Gdn_Format::To($Values['Plugin.Signatures.Sig'], $Sender->Form->GetFormValue('Format'));
```
Comment (tburry) [inline plugins/Signatures/class.signatures.plugin.php]: This is odd code. If the format isn't coming in `$Values` then it should. You also need to default the format to `C('Garden.InputFormatter')`.

Comment (tburry) [inline plugins/Signatures/class.signatures.plugin.php]: Also you can't wrong with `val('Plugin.Signatures.Sig', $Values)`.

**Pull request merged**